### PR TITLE
Experimenting with Recent Menu

### PR DESCRIPTION
**Pull request by sbatten**
@bpasero @isidorn 
Experimenting with ways to remove custom logic for the recent menu in the menu bar. I don't know where this code would live, but the idea is that a listener manages the Recent Submenu data model when the recent menu is updated. This would mean commands would have to be registered to handle the different URIs. Not sure if this can be parameterized to have a single command that can be re-used. This would both remove custom logic from the menu bar but I *think* we could reuse it in the native case as well. Not sure due to fallback case on macOS.

P.S. don't try this code with the native menu
/cc @jrieken is this menu model abuse?

### CODE REVIEW

**Comment by bpasero** [comment]
@sbatten yeah I think direction is what I had in mind as well. Instead of generating a lot of random commands it would be nicer if you could express some kind of argument that should get passed to a command (the URI to open).

**Comment by jrieken** [comment]
> /cc @jrieken is this menu model abuse?

Not sure that I can follow you... Can you point me to where things happen?

**Comment by sbatten** [comment]
@jrieken essentially, every time the recently opened files changes, we will dispose of these menu items (up to 10) and add the new set with MenuRegistry.appendMenu

File: src/vs/workbench/browser/parts/titlebar/menubarControl.ts [hunk 1]
```diff
@@ -422,7 +490,7 @@ export class MenubarControl extends Disposable {
 	private insertActionsBefore(nextAction: IAction, target: IAction[]): void {
 		switch (nextAction.id) {
 			case 'workbench.action.openRecent':
-				target.push(...this.getOpenRecentActions());
+				// target.push(...this.getOpenRecentActions());
```
Comment (umartinez22) [inline src/vs/workbench/browser/parts/titlebar/menubarControl.ts]: If you don't use this line, could you remove this?

**Comment by sbatten** [comment]
@bpasero latest commit shows what that might look like. I chose to add arguments to the menu registration (as opposed to forwarding args through the menu widget) because otherwise we just end up doing the same thing in a different way. 

/cc @jrieken for changes to MenuRegistry and MenuItemAction

**Pull request closed without merging**